Firestore/Source/Public/FIRDocumentSnapshot.h

```diff
@@ -46,21 +46,44 @@ NS_SWIFT_NAME(DocumentSnapshot)
 @property(nonatomic, strong, readonly) FIRSnapshotMetadata *metadata;
 
 /**
- * Retrieves all fields in the document as an `NSDictionary`.
+ * Retrieves all fields in the document as an `NSDictionary`. Returns `nil` if the document doesn't
+ * exist.
  *
- * @return An `NSDictionary` containing all fields in the document.
+ * @return An `NSDictionary` containing all fields in the document or `nil` if the document doesn't
+ *      exist.
  */
-- (NSDictionary<NSString *, id> *)data;
+- (nullable NSDictionary<NSString *, id> *)data;
 
 /**
  * Retrieves a specific field from the document.
  *
  * @param key The field to retrieve.
  *
- * @return The value contained in the field or `nil` if the field doesn't exist.
+ * @return The value contained in the field or `nil` if the document or field doesn't exist.
  */
 - (nullable id)objectForKeyedSubscript:(id)key;
 
 @end
 
+/**
+ * A `FIRQueryDocumentSnapshot` contains data read from a document in your Firestore database. The
```

Comment:
Perhaps "contains data read from a document in your Firestore database as a part of a query."

Because these types are related we should add clarifying text here that describes when the two types are used, i.e. `DocumentSnapshot` for single document gets and listens, `QueryDocumentSnapshot` for query/collection gets/listens.

Reply:
Changed the comments. Let me know what you think since I need to backport them to the other platforms.